functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
+{
+<#
+.SYNOPSIS 
+Restores a SQL Server Database from a set of backupfiles
+
+.DESCRIPTION
+Scans a given folder for Full, Differential and Log backups. 
+OR
+Takes a set of folder arrays and processes them for Full, Differential and Log backups.
+
+These are then filtered and restored to a specified SQL Server intance and file location
+
+The backup LSN chain and RecoveryForkID will also be checked to ensure the restore is valid
+
+It can also generate restore scripts, both as part of a restore or as it's only action
+
+.PARAMETER SqlServer
+The SQL Server instance. 
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Path
+Path to SQL Server backup file. This can be a full, differential or log backup file.
+
+.PARAMETER Files
```

Comment:
There is one exception to the singular rule -- which is why Sync-SqlLoginPermissions is plural -- when the ..actually, I'l just paste.

From "[Strongly Encouraged Development Guidelines](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx)"
Use Singular Parameter Names
Avoid using plural names for parameters whose value is a single element. This includes parameters that take arrays or lists because the user might supply an array or list with only one element.

Plural parameter names should be used only in those cases where the value of the parameter is always a multiple-element value. In these cases, the cmdlet should verify that multiple elements are supplied, and the cmdlet should display a warning to the user if multiple elements are not supplied.

_______________
If it's always plural, then it can be plural, but it looks like since there is an (s), that you can pass a singular. Consider FileList instead? (@powerdbaklaas) will probably disagree here ;)